scipy/stats/tests/test_morestats.py

```diff
@@ -697,39 +723,104 @@ def test_confidence_intervals2(self, alternative, pval, ci_low, ci_high):
         res = stats.binomtest(3, n=50, p=0.2, alternative=alternative)
         assert_allclose(res.pvalue, pval, rtol=1e-6)
         assert_equal(res.proportion_estimate, 0.06)
-        proportion_ci = res.proportion_ci(confidence_level=0.99)
-        assert_allclose(proportion_ci, (ci_low, ci_high), rtol=1e-6)
+        ci = res.proportion_ci(confidence_level=0.99)
+        assert_allclose((ci.low, ci.high), (ci_low, ci_high), rtol=1e-6)
+
+    # Expected results here are from R 3.6.2 binom.test.
+    @pytest.mark.parametrize('alternative, pval, ci_high',
+                             [('less', 0.05631351, 0.2588656),
+                              ('greater', 1.0, 1.0),
+                              ('two-sided', 0.07604122, 0.3084971)])
+    def test_confidence_interval_exact_k0(self, alternative, pval, ci_high):
+        # Test with k=0, n = 10.
+        res = stats.binomtest(0, 10, p=0.25, alternative=alternative)
+        assert_allclose(res.pvalue, pval, rtol=1e-6)
+        ci = res.proportion_ci(confidence_level=0.95)
+        assert_equal(ci.low, 0.0)
+        assert_allclose(ci.high, ci_high, rtol=1e-6)
+
+    # Expected results here are from R 3.6.2 binom.test.
+    @pytest.mark.parametrize('alternative, pval, ci_low',
+                             [('less', 1.0, 0.0),
+                              ('greater', 9.536743e-07, 0.7411344),
+                              ('two-sided', 9.536743e-07, 0.6915029)])
+    def test_confidence_interval_exact_k_is_n(self, alternative, pval, ci_low):
+        # Test with k = n = 10.
+        res = stats.binomtest(10, 10, p=0.25, alternative=alternative)
+        assert_allclose(res.pvalue, pval, rtol=1e-6)
+        ci = res.proportion_ci(confidence_level=0.95)
+        assert_equal(ci.high, 1.0)
+        assert_allclose(ci.low, ci_low, rtol=1e-6)
 
     # Expected results are from the prop.test function in R 3.6.2.
     @pytest.mark.parametrize(
-        'alternative, corr, conf, ci_low, ci_high',
-        [['two-sided', True, 0.95, 0.08094782, 0.64632928],
-         ['two-sided', True, 0.99, 0.0586329, 0.7169416],
-         ['two-sided', False, 0.95, 0.1077913, 0.6032219],
-         ['two-sided', False, 0.99, 0.07956632, 0.6799753],
-         ['less', True, 0.95, 0.0, 0.6043476],
-         ['less', True, 0.99, 0.0, 0.6901811],
-         ['less', False, 0.95, 0.0, 0.5583002],
-         ['less', False, 0.99, 0.0, 0.6507187],
-         ['greater', True, 0.95, 0.09644904, 1.0],
-         ['greater', True, 0.99, 0.06659141, 1.0],
-         ['greater', False, 0.95, 0.1268766, 1.0],
-         ['greater', False, 0.99, 0.08974147, 1.0]]
+        'k, alternative, corr, conf, ci_low, ci_high',
```

Comment:
minor comment, these could be 4 stacked `@pytest.mark.parametrize`s to simplify a bit but they're ok as is

Reply:
For the given inputs, yes, this is basically a nested loop (well, we'd need different tests for the inputs exactly as given, but additional values for the 0.99 confidence level could be added to make it a big nested loop over `k`, `alternative`, `corr` and `conf`), but the expected results are different in each case.  This is verbose, but it puts the expected values right next to the corresponding inputs.